app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: %i[show destroy]
+
+  def new; end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
+    if @post.save
+      flash[:success] = 'Post successfully created'
+    else
+      flash[:danger] = 'Oops, Your post is not valid!'
+    end
+    redirect_to posts_path
+  end
+
+  def show
+    @post = Post.find(params[:id])
+    @comment = Comment.new
+  end
+
+  def index
+    @user = current_user
+    @post = Post.new
+    @posts = Post.all
+  end
+
+  private
```

Comment:
- [ ] **[OPTIONAL]** We don't have to send @user to view, as `current_user` is already available to all views by default.

Reply:
fixed, thank you

Reply:
please disregard the unresolved merging conflicts, it will be fixed manually by console